### PR TITLE
feat(normalization): Reject stale events

### PR DESCRIPTION
**Pull request by iker-barriocanal**
There are some cases where events with stale timestamps reach Relay. Relay doesn't perform full normalization on these and forwards them, causing issues in later parts of the pipeline. This PR introduces a new processor to deal with these timestamps, rejecting events with old timestamps as a first step.

### CODE REVIEW

File: relay-event-normalization/src/normalize/mod.rs [hunk 1]
```diff
@@ -1021,6 +1022,8 @@ pub fn light_normalize_event(
         // Check for required and non-empty values
         schema::SchemaProcessor.process_event(event, meta, ProcessingState::root())?;
 
+        TimestampProcessor.process_event(event, meta, ProcessingState::root())?;
```
Comment (iker-barriocanal) [inline relay-event-normalization/src/normalize/mod.rs:1024]: The timestamp processor runs before the clock drift processor. With this PR, Relay will drop events created before and coming from systems with a date before January 1, 1970. I believe it's acceptable to assume no systems will have their date set to a value prior to 1970.

File: relay-event-normalization/src/timestamp.rs [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    ProcessingAction, ProcessingResult, ProcessingState, Processor,
+};
+use relay_event_schema::protocol::Event;
+use relay_protocol::Meta;
+
+/// Ensures an event's timestamps are not stale.
+///
+/// Stale timestamps are those that happened before January 1, 1970 UTC. The
+/// processor validates the start and end timestamps of an event and returns an
+/// error if any of these is stale.
```
Comment (iker-barriocanal) [inline relay-event-normalization/src/timestamp.rs]: This will evolve as this processor checks for more values.

Comment (TBS1996) [inline relay-event-normalization/src/timestamp.rs]:
```suggestion
/// error if any of these are stale.
```

(i feel so petty doing this)

**Review by TBS1996 — APPROVED** [review]
looks good to me

**Pull request merged**